src/main/java/org/spongepowered/api/service/command/CommandService.java

```diff
@@ -44,72 +45,90 @@
+     * <p>If there is a conflict with one of the aliases (i.e. that alias is
+     * already assigned to another command), and this CommandService does not
+     * allow an alias to be registered twice , then the alias will be skipped.
+     * It is possible for there to be no alias to be available out of the
+     * provided list of aliases, which would mean that the command would not be
+     * assigned to any aliases.</p>
+     * 
      *
      * <p>The first non-conflicted alias becomes the "primary alias."</p>
      *
-     * @param plugin A plugin instance
+     * @param registrar The object registering the command. Can be a class
+     *        annotated by {@link Plugin}, or anything implementing
+     *        {@link CommandRegistrar}.
      * @param callable The command
      * @param alias An array of aliases
-     * @return The registered command mapping, unless no aliases could be registered
-     * @throws IllegalArgumentException Thrown if {@code plugin} is not a plugin instance
+     * @return The registered command mapping, unless no aliases could be
+     *         registered
+     * @throws IllegalArgumentException Thrown if {@code plugin} is not a plugin
+     *         instance
+     * @throws IllegalArgumentException Thrown if a command name is invalid.
      */
-    Optional<CommandMapping> register(Object plugin, CommandCallable callable, String... alias);
+    Optional<CommandMapping> register(Object registrar, CommandCallable callable, String... alias);
 
     /**
      * Register a given command using the given list of aliases.
      *
-     * <p>If there is a conflict with one of the aliases (i.e. that alias
-     * is already assigned to another command), then the alias will be skipped.
-     * It is possible for there to be no alias to be available out of
-     * the provided list of aliases, which would mean that the command would not
-     * be assigned to any aliases.</p>
+     * 
+     * <p>If there is a conflict with one of the aliases (i.e. that alias is
+     * already assigned to another command), and this CommandService does not
+     * allow an alias to be registered twice , then the alias will be skipped.
+     * It is possible for there to be no alias to be available out of the
+     * provided list of aliases, which would mean that the command would not be
+     * assigned to any aliases.</p>
+     * 
      *
      * <p>The first non-conflicted alias becomes the "primary alias."</p>
      *
-     * @param plugin A plugin instance
+     * @param registrar The object registering the command. Can be a class
+     *        annotated by {@link Plugin}, or anything implementing
+     *        {@link CommandRegistrar}.
      * @param callable The command
-     * @param aliases A list of aliases
-     * @return The registered command mapping, unless no aliases could be registered
-     * @throws IllegalArgumentException Thrown if {@code plugin} is not a plugin instance
+     * @param alias An array of aliases
```

Comment:
No reason to change "list" to "array" here. In fact, 'list' is more correct because it's `List<String>` not `String[]`